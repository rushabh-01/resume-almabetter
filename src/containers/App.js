import React,{useState,useEffect} from "react";
import Form from "../components/Pages/DetailPage/Form";
import Navbar from '../components/Navbar';
import "./styles.css";
import ResumeContextProvider from "../contexts/ResumeContext";
import {Route, Switch, Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import About from "../components/Pages/About";
import styled from "styled-components";
import Animation from "../components/Animation";
import { Button } from "@material-ui/core";
import Footer from "../components/Footer";

const skinCodes = ['skin1', 'skin2', 'skin3','skin4' ]; 

function Templates() {
  const useStyles = makeStyles({
    headerLink: {
      color: "black",
      minWidth: 100,
      position:"relative",
      margin:15,
      background:"white",
      opacity:0.5,
    },
  });

  const classes = useStyles();

  const [isShown, setIsShown] = useState(false);

  return (
    <div className="left">
      <h2 style={{textAlign:'center'}}>Build your brand-new resume in as little as 5 minutes.</h2>
      <h4 style={{textAlign:'center'}}>Chosse a template</h4>
      <Card>
      {skinCodes.map((value,index) =>(
          <Link to='/detail/info' key={index}>
            <TemplateCard  onMouseEnter={() => setIsShown(index)} onMouseLeave={() => setIsShown(false)}>
                <img src={value + '.svg'} alt="" className="templateImage"/>
                {isShown === index && (<Button variant="contained" className={classes.headerLink}>use template</Button>)}
            </TemplateCard>
          </Link>
        ))}
      </Card>
    </div>
  );
}

function App() {

  const [mobile, setMobile] = useState(true);
  useEffect(() => {
    window.innerWidth < 768 ? setMobile(true) : setMobile(false);
  }, [mobile]);

  return (
    
    <div className="app">
      <Animation/>
      <ResumeContextProvider>
      
        
        <Navbar/>
        
          <Switch>
            <Route path="/" component={Templates} exact />
            <Route path="/detail" component={Form} />
            <Route path="/about" component={About} />
          </Switch>

          {mobile ? <Footer/> : "" }
        
      </ResumeContextProvider>
    
      
      </div>
  );
}

const Card = styled.div`
display: flex;
margin: 25px;
justify-content: space-evenly;
flex-wrap: wrap;
`
const TemplateCard = styled.div`
display: flex;
flex-direction: column;
align-items: center;
@media (max-width:768px){
  margin:20px 0;
}
`



export default App;
