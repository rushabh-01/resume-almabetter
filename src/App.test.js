import { render, screen } from '@testing-library/react';
import App from './containers/App';
import {BrowserRouter} from 'react-router-dom';

// unit testing

test('renders', () => {
  render(
    <BrowserRouter><App/></BrowserRouter>)
  const text = screen.getByText('Build your brand-new resume in as little as 5 minutes.');
  expect(text).toBeInTheDocument();
});

// snapshot testing

it("should take a snapshot", () => {
	const { asFragment } = render(<BrowserRouter><App/></BrowserRouter>);

	expect(asFragment(<App />)).toMatchSnapshot();
});





