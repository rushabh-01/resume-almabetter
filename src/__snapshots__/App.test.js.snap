// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      style="position: fixed; top: 0px; width: 100vw; height: 100vh;"
    >
      <div
        class="sc-eDWCr jHmALf"
      >
        <div
          class="sc-bqWxrE bBgdDi"
        >
          <li />
          <li />
          <li />
          <li />
          <li />
          <li />
          <li />
          <li />
          <li />
          <li />
        </div>
      </div>
    </div>
    <div
      class="sc-jrcTuL cMVRsC navbar"
    >
      <a
        class="home-link"
        href="/"
      >
        <div
          class="sc-ipEyDJ dmmAtF"
        >
          <p
            class="logo"
          >
            ResumePlus
          </p>
        </div>
      </a>
      <div
        class="sc-kDvujY feCCco"
      >
        <a
          href="/"
        >
          Resume templates
        </a>
        <a
          href="/about"
        >
          About us
        </a>
      </div>
    </div>
    <div
      class="left"
    >
      <h2
        style="text-align: center;"
      >
        Build your brand-new resume in as little as 5 minutes.
      </h2>
      <h4
        style="text-align: center;"
      >
        Chosse a template
      </h4>
      <div
        class="sc-hBxehG ihQuus"
      >
        <a
          href="/detail/info"
        >
          <div
            class="sc-fnGiBr eXAihs"
          >
            <img
              alt=""
              class="templateImage"
              src="skin1.svg"
            />
          </div>
        </a>
        <a
          href="/detail/info"
        >
          <div
            class="sc-fnGiBr eXAihs"
          >
            <img
              alt=""
              class="templateImage"
              src="skin2.svg"
            />
          </div>
        </a>
        <a
          href="/detail/info"
        >
          <div
            class="sc-fnGiBr eXAihs"
          >
            <img
              alt=""
              class="templateImage"
              src="skin3.svg"
            />
          </div>
        </a>
        <a
          href="/detail/info"
        >
          <div
            class="sc-fnGiBr eXAihs"
          >
            <img
              alt=""
              class="templateImage"
              src="skin4.svg"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
